src/lock.rs

```diff
@@ -293,3 +294,60 @@ impl From<&LockedPackage> for FileRequirement {
         )
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::{collections::BTreeMap, str::FromStr};
+
+    use semver::Version;
+
+    use crate::{package::PackageName, registry::RegistryUri};
+
+    use super::{Digest, DigestAlgorithm, FileRequirement, LockedPackage, Lockfile};
+
+    #[test]
+    fn stable_file_requirement_order() {
+        let lock = Lockfile {
+            packages: BTreeMap::from([
+                (
+                    PackageName::new("package1").unwrap(),
+                    LockedPackage {
+                        name: PackageName::new("package1").unwrap(),
+                        digest: Digest::from_parts(
+                            DigestAlgorithm::SHA256,
+                            "c109c6b120c525e6ea7b2db98335d39a3272f572ac86ba7b2d65c765c353c122",
+                        )
+                        .unwrap(),
+                        registry: RegistryUri::from_str("http://my-registry.com").unwrap(),
+                        repository: "my-repo".to_owned(),
+                        version: Version::new(0, 1, 0),
+                        dependencies: vec![],
+                        dependants: 1,
+                    },
+                ),
+                (
+                    PackageName::new("package2").unwrap(),
+                    LockedPackage {
+                        name: PackageName::new("package2").unwrap(),
+                        digest: Digest::from_parts(
+                            DigestAlgorithm::SHA256,
+                            "c109c6b120c525e6ea7b2db98335d39a3272f572ac86ba7b2d65c765c353bce3",
+                        )
+                        .unwrap(),
+                        registry: RegistryUri::from_str("http://my-registry.com").unwrap(),
+                        repository: "my-other-repo".to_owned(),
+                        version: Version::new(0, 2, 0),
+                        dependencies: vec![],
+                        dependants: 1,
+                    },
+                ),
+            ]),
+        };
+
+        let files: Vec<FileRequirement> = lock.clone().into();
+        for _ in 0..30 {
+            let other_files: Vec<FileRequirement> = lock.clone().into();
```

Comment:
Im not sure that you are actually testing anything here. You are cloning which is a usually byte by byte memcpy - I believe you would persist any initial random ordering occurred from building a HashMap.

https://docs.rs/hashbrown/latest/src/hashbrown/map.rs.html#190-204

https://docs.rs/hashbrown/latest/src/hashbrown/raw/mod.rs.html#3238

I can't verify right now, but does this test fail if you plug in a HashMap instead of a BTreeMap? 

I think testing the ordering a little more explicit here would be good! Maybe using two different instances of the struct instead of cloning it (which would also get rid of the cfg_attr attribute)